ogr/ogrsf_frmts/gml/ogrgmldatasource.cpp

```diff
@@ -2715,7 +2715,7 @@ void OGRGMLDataSource::InsertHeader()
                      strcmp(poFieldDefn->GetNameRef(), "fid") == 0)
                 continue;
 
-            const auto AddComment = [fpSchema, poFieldDefn]()
+            const auto AddComment = [this, fpSchema, poFieldDefn]()
```

Comment:
you added `this' but the lambda body does not seem to have changed. Why is it now necessary?

Reply:
This is due to the OGRGMLDataSource::PrintLine() method that has been changed from static to non-static and this lambda calls it